service/modules/product/repository/product.go

```diff
@@ -0,0 +1,55 @@
+package repository
+
+import (
+    "github.com/ricnah/workit-be/service/extensions/terror"
+    "github.com/ricnah/workit-be/types/models"
+    "gorm.io/gorm"
+)
+
+type productRepository struct {
+    db *gorm.DB
+}
+
+func NewProductRepository(db *gorm.DB) ProductRepository {
+    return &productRepository{db: db}
+}
+
+func (r *productRepository) CreateProduct(product *models.Product) (terr terror.ErrInterface) {
+    err := r.db.Create(product).Error
+    if err != nil {
+        terr = terror.New(err)
+    }
+    return
+}
+
+func (r *productRepository) GetProducts() (products []models.Product, terr terror.ErrInterface) {
+    err := r.db.Find(&products).Error
+    if err != nil {
+        terr = terror.New(err)
+    }
+    return
+}
+
+func (r *productRepository) GetProductByID(id int64) (product models.Product, terr terror.ErrInterface) {
```

Comment:
yang konsepnya get repo db, ditambahin cek gorm data not found. error itu ada beberapa macem, misal error karena koneksi ke db error, error sintax mysql yang lebih ke kesalahan infrastruktur. sedangkan code karena datanya ga ada di database itu lebih ke bisnis logic. handlingnya nanti bisa beda.
contoh di module user UserGetByID